pcsx2/GS/Renderers/Vulkan/GSTextureVK.cpp

```diff
@@ -26,134 +26,246 @@
-					VK_IMAGE_USAGE_TRANSFER_SRC_BIT | VK_IMAGE_USAGE_TRANSFER_DST_BIT |
+			ici.usage = VK_IMAGE_USAGE_TRANSFER_SRC_BIT | VK_IMAGE_USAGE_TRANSFER_DST_BIT |
 						VK_IMAGE_USAGE_COLOR_ATTACHMENT_BIT | VK_IMAGE_USAGE_SAMPLED_BIT |
-						VK_IMAGE_USAGE_INPUT_ATTACHMENT_BIT))
-			{
-				return {};
-			}
-
-			Vulkan::SetObjectName(
-				g_vulkan_context->GetDevice(), texture.GetImage(), "%ux%u render target", width, height);
-			return std::make_unique<GSTextureVK>(type, format, std::move(texture));
+						VK_IMAGE_USAGE_INPUT_ATTACHMENT_BIT;
 		}
+		break;
 
 		case Type::DepthStencil:
 		{
 			pxAssert(levels == 1);
-
-			VKTexture texture;
-			if (!texture.Create(width, height, levels, 1, vk_format, VK_SAMPLE_COUNT_1_BIT,
-					VK_IMAGE_VIEW_TYPE_2D, VK_IMAGE_TILING_OPTIMAL,
-					VK_IMAGE_USAGE_TRANSFER_SRC_BIT | VK_IMAGE_USAGE_TRANSFER_DST_BIT |
-						VK_IMAGE_USAGE_DEPTH_STENCIL_ATTACHMENT_BIT | VK_IMAGE_USAGE_SAMPLED_BIT))
-			{
-				return {};
-			}
-
-			Vulkan::SetObjectName(
-				g_vulkan_context->GetDevice(), texture.GetImage(), "%ux%u depth stencil", width, height);
-			return std::make_unique<GSTextureVK>(type, format, std::move(texture));
+			ici.usage = VK_IMAGE_USAGE_TRANSFER_SRC_BIT | VK_IMAGE_USAGE_TRANSFER_DST_BIT |
+						VK_IMAGE_USAGE_DEPTH_STENCIL_ATTACHMENT_BIT | VK_IMAGE_USAGE_SAMPLED_BIT;
+			vci.subresourceRange.aspectMask = VK_IMAGE_ASPECT_DEPTH_BIT;
 		}
+		break;
 
 		case Type::RWTexture:
 		{
 			pxAssert(levels == 1);
+			ici.usage = VK_IMAGE_USAGE_TRANSFER_SRC_BIT | VK_IMAGE_USAGE_TRANSFER_DST_BIT | VK_IMAGE_USAGE_STORAGE_BIT |
+						VK_IMAGE_USAGE_SAMPLED_BIT;
+		}
+		break;
+
+		default:
+			return {};
+	}
+
+	// Use dedicated allocations for typical RT size
+	if ((type == Type::RenderTarget || type == Type::DepthStencil) && width >= 512 && height >= 448)
+		aci.flags |= VMA_ALLOCATION_CREATE_DEDICATED_MEMORY_BIT;
+
+	VkImage image = VK_NULL_HANDLE;
+	VmaAllocation allocation = VK_NULL_HANDLE;
+	VkResult res = vmaCreateImage(g_vulkan_context->GetAllocator(), &ici, &aci, &image, &allocation, nullptr);
+	if (aci.flags & VMA_ALLOCATION_CREATE_DEDICATED_MEMORY_BIT && res != VK_SUCCESS)
+	{
+		// try without dedicated allocation
+		aci.flags &= ~VMA_ALLOCATION_CREATE_DEDICATED_MEMORY_BIT;
+		res = vmaCreateImage(g_vulkan_context->GetAllocator(), &ici, &aci, &image, &allocation, nullptr);
+	}
+	if (res == VK_ERROR_OUT_OF_DEVICE_MEMORY)
+	{
+		DevCon.WriteLn("Failed to allocate device memory for %ux%u texture", width, height);
+		return {};
+	}
+	else if (res != VK_SUCCESS)
+	{
+		LOG_VULKAN_ERROR(res, "vmaCreateImage failed: ");
+		return {};
+	}
+
+	VkImageView view = VK_NULL_HANDLE;
+	vci.image = image;
+	res = vkCreateImageView(g_vulkan_context->GetDevice(), &vci, nullptr, &view);
+	if (res != VK_SUCCESS)
+	{
+		LOG_VULKAN_ERROR(res, "vkCreateImageView failed: ");
+		vmaDestroyImage(g_vulkan_context->GetAllocator(), image, allocation);
+		return {};
+	}
+
+	switch (type)
```

Comment:
CI warning
`enumeration value 'Invalid' not handled in switch [-Wswitch]`

Reply:
False positive as usual.. tbh that warning annoys me enough that I'd almost consider turning it off.